### PR TITLE
downgrade OpenMPI to 3.1.1

### LINKED ISSUE #523: Avoid shared memory bug in openmpi>=3.1.2
Currently we build v3.1.2 in `nsls2-tag`:
https://github.com/NSLS-II/lightsource2-recipes/blob/87723db7b7d8c3b7475b1e478ac1a394326dec48/recipes-tag/openmpi/meta.yaml#L2
However, this version seems to be buggy. If one spawns a few MPI processes, let them do some work, but terminate them abnormally (`Ctrl-C` and whatnot), it can be seen that in `/dev/shm/` there will be shared memory segments related to openmpi's `vader` component (don't ask me what this is...) that are not unlinked by openmpi during the cleanup phase:
```bash
leofang@xf03id-srv5:~$ ls -lt /dev/shm | more
total 28464
-rw------- 1 leofang leofang 4194312 Jan 25 23:06 vader_segment.xf03id-srv5.73dc0001.2
-rw------- 1 leofang leofang 4194312 Jan 25 23:06 vader_segment.xf03id-srv5.73dc0001.3
-rw------- 1 leofang leofang 4194312 Jan 25 23:06 vader_segment.xf03id-srv5.73dc0001.1
-rw------- 1 leofang leofang 4194312 Jan 25 23:06 vader_segment.xf03id-srv5.73dc0001.0
-rw------- 1 xjhuang xjhuang 4194312 Jan 15 11:08 vader_segment.xf03id-srv5.32270001.2
-rw------- 1 xjhuang xjhuang 4194312 Jan 15 11:08 vader_segment.xf03id-srv5.32270001.0
-rw------- 1 xjhuang xjhuang 4194312 Jan 15 11:08 vader_segment.xf03id-srv5.32270001.3
-rw------- 1 xjhuang xjhuang 4194312 Jan 15 11:08 vader_segment.xf03id-srv5.32270001.1
-rw------- 1 xjhuang xjhuang 4194312 Jan 15 11:07 vader_segment.xf03id-srv5.358e0001.3
-rw------- 1 xjhuang xjhuang 4194312 Jan 15 11:07 vader_segment.xf03id-srv5.358e0001.2
......
``` 
and they will remain there until the system is reboot, eating up slowly the system's memory! 

Based on openmpi's changelog (see [here](https://raw.githubusercontent.com/open-mpi/ompi/v3.1.x/NEWS)), it seems that `vader` was reworked in v3.1.2, presumably this bug sneaked in by then. ~~There's a bug fix in v3.1.3 hopefully would address this~~; if not, we can downgrade to v3.1.1, which I tested and worked without this issue. 

(**UPDATE**: v3.1.3 also has this problem, has to use 3.1.1...)

I have other questions related to building conda packages for `mpi4py`, `openmpi`, and `mpich`, but perhaps I should ask them somewhere else...

### PR DESCRIPTION
**Pull request by leofang**
closes #523. @mrakitin when this PR is merged, please remove any Open MPI we built previously from Anaconda. Thanks!

### CODE REVIEW

**Comment by leofang** [comment]
Note that mpi4py also needs to be rebuilt against this version (but I think we don't have any, do we?)

**Comment by mrakitin** [comment]
@leofang, let's address it after #559. I'll look at it today, sorry for the delay.

**Comment by leofang** [comment]
Agree, I meant to say this too. Thanks @mrakitin!

File: recipes-tag/openmpi/meta.yaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
-# copied and pasted by Leo Fang from the following commit:
+# adapted by Leo Fang from the following commit:
 # https://github.com/conda-forge/openmpi-feedstock/blob/1d6390794529ad80f5b3416fa2cb98882d1c95fa/recipe/meta.yaml
 
-{% set version = "3.1.3" %}
+{% set version = "3.1.1" %}
 {% set sha256 = "8be04307c00f51401d3fb9d837321781ea7c79f2a5a4a2e5d4eaedc874087ab6" %}
 {% set major = version.rpartition('.')[0] %}
-{% set build = 1001 %}
+{% set build = 1000 %}
```
Comment (mrakitin) [inline recipes-tag/openmpi/meta.yaml]: Set the build number to `1000` here as we don't have any build for openmpi (I manually removed them).

File: recipes-tag/openmpi/meta.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
-# copied and pasted by Leo Fang from the following commit:
+# adapted by Leo Fang from the following commit:
 # https://github.com/conda-forge/openmpi-feedstock/blob/1d6390794529ad80f5b3416fa2cb98882d1c95fa/recipe/meta.yaml
 
-{% set version = "3.1.3" %}
+{% set version = "3.1.1" %}
 {% set sha256 = "8be04307c00f51401d3fb9d837321781ea7c79f2a5a4a2e5d4eaedc874087ab6" %}
```
Comment (mrakitin) [inline recipes-tag/openmpi/meta.yaml]: Ups, this caused the issue:
```
ERROR:build.py:SHA256 mismatch: '3f11b648dd18a8b878d057e9777f2c43bf78297751ad77ae2cef6db0fe80c77c' != '8be04307c00f51401d3fb9d837321781ea7c79f2a5a4a2e5d4eaedc874087ab
```

**Pull request merged**